Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -435,41 +427,45 @@ theorem lift_power (a b : Cardinal.{u}) : lift.{v} (a ^ b) = lift.{v} a ^ lift.{
     mk_congr <| Equiv.ulift.trans (Equiv.ulift.arrowCongr Equiv.ulift).symm
 
 @[simp]
-theorem power_zero {a : Cardinal} : a ^ (0 : Cardinal) = 1 :=
+theorem power_zero (a : Cardinal) : a ^ (0 : Cardinal) = 1 :=
   inductionOn a fun _ => mk_eq_one _
 
 @[simp]
-theorem power_one {a : Cardinal.{u}} : a ^ (1 : Cardinal) = a :=
+theorem power_one (a : Cardinal.{u}) : a ^ (1 : Cardinal) = a :=
   inductionOn a fun α => mk_congr (Equiv.funUnique (ULift.{u} (Fin 1)) α)
 
-theorem power_add {a b c : Cardinal} : a ^ (b + c) = a ^ b * a ^ c :=
+theorem power_add (a b c : Cardinal) : a ^ (b + c) = a ^ b * a ^ c :=
   inductionOn₃ a b c fun α β γ => mk_congr <| Equiv.sumArrowEquivProdArrow β γ α
 
+private theorem cast_succ (n : ℕ) : ((n + 1 : ℕ) : Cardinal.{u}) = n + 1 := by
+  change #(ULift.{u} _) = #(ULift.{u} _) + 1
```

Comment:
Out of scope if you want it to be, but it would be nice to have a `natCast_def` lemma that avoids the need for this `change`.

Reply:
This wouldn't be very useful, as it'd be essentially subsumed by [`Cardinal.mk_fintype`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Cardinal/Basic.html#Cardinal.mk_fintype).

Reply:
The same could be said for `add_def` vs `mk_sum` though, right?

Reply:
Not quite, since one is universe polymorphic, while the other is more convenient in the case the universes are equal.

`#(Ulift (Fin n)) = n` is just `mk_ulift` composed with `mk_fintype`.

Reply:
In fact, we should probably redefine `↑n` as the def-eq `lift #(Fin n)`

Reply:
```lean
theorem Cardinal.natCast_def (n : ℕ) : (n : Cardinal.{u}) = #(ULift.{u} (Fin n)) := by
  rw [mk_uLift, mk_fin, lift_natCast]
```
is probably the steps you're thinking of; but I think I'd still argue that the lemma is worth having since it's `rfl` and avoids a `change`. No need to pull this PR back from bors though.